tools/Microsoft.TemplateEngine.Authoring.TemplateApiVerifier/TemplateVerifierOptionsExtensions.cs

```diff
@@ -0,0 +1,176 @@
+            this TemplateVerifierOptions options,
+            IReadOnlyDictionary<string, string?>? inputParameters)
+        {
+            return options.WithCustomInstatiation(async verifierOptions => await RunInstantiation(verifierOptions, inputParameters, null).ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Adds custom template instantiator that runs template instantiation in-proc via TemplateCreator API.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="inputDataSet"></param>
+        /// <returns></returns>
+        public static TemplateVerifierOptions WithInstantiationThroughTemplateCreatorApi(
+            this TemplateVerifierOptions options,
+            InputDataSet? inputDataSet)
+        {
+            return options.WithCustomInstatiation(async verifierOptions => await RunInstantiation(verifierOptions, null, inputDataSet).ConfigureAwait(false));
+        }
+
+        private static async Task<IInstantiationResult> RunInstantiation(
+            TemplateVerifierOptions options,
+            IReadOnlyDictionary<string, string?>? inputParameters,
+            InputDataSet? inputDataSet)
+        {
+            if (!string.IsNullOrEmpty(options.DotnetExecutablePath))
+            {
+                throw new TemplateVerificationException(LocalizableStrings.Error_DotnetPath, TemplateVerificationErrorCode.InvalidOption);
+            }
+
+            if (string.IsNullOrEmpty(options.TemplatePath))
+            {
+                throw new TemplateVerificationException(LocalizableStrings.Error_TemplatePathMissing, TemplateVerificationErrorCode.InvalidOption);
+            }
+
+            if (!Path.Exists(options.TemplatePath))
+            {
+                throw new TemplateVerificationException(LocalizableStrings.Error_TemplatePathDoesNotExist, TemplateVerificationErrorCode.InvalidOption);
+            }
+
+            if (options.TemplateSpecificArgs != null)
+            {
+                throw new TemplateVerificationException(LocalizableStrings.Error_TemplateArgsDisalowed, TemplateVerificationErrorCode.InvalidOption);
+            }
+
+            string templateConfigPath = options.TemplatePath;
+            if (!templateConfigPath.EndsWith(ExpectedConfigLocation, StringComparison.InvariantCultureIgnoreCase))
+            {
+                templateConfigPath = Path.Combine(templateConfigPath, ExpectedConfigLocation);
+
+                if (!File.Exists(templateConfigPath))
+                {
+                    throw new TemplateVerificationException(
+                        string.Format(LocalizableStrings.Error_ConfigDoesntExist, templateConfigPath, ExpectedConfigLocation),
+                        TemplateVerificationErrorCode.InvalidOption);
+                }
+            }
+
+            // the expected nesting is checked and ensured above
+            string templateBasePath = new FileInfo(templateConfigPath).Directory!.Parent!.FullName;
```

Comment:
I would consider using full path here:
- install template \ use custom provider to load template (using `TemplatePackageManager`)
- get required template (using `TemplatePackageManager`)
- create it (using `TemplateCreator`)

It would probably be also a cleaner code. Suggesting to force using virtual hive to avoid clashes. `IDE.Bootstrapper` can be used instead.
Current approach is very specific and won't work for NuGet packages for example and also error-prone.

Reply:
Yhank you for bringing this to my attention! I like this!